pkg/cluster/internal/providers/docker/provision.go

```diff
@@ -294,6 +294,9 @@ func getProxyEnv(cfg *config.Cluster, networkName string, nodeNames []string) (m
 
 		noProxyList := append(subnets, envs[common.NOProxy])
 		noProxyList = append(noProxyList, nodeNames...)
+		// Add .svc explicitly to no_proxy to allow in cluster
+		// pod and service dns resolution
+		noProxyList = append(noProxyList, ".svc")
```

Comment:
I think that we have to no_proxy the cluster domain , if we look at the standard coredns config 
```yaml
apiVersion: v1
data:
  Corefile: |
    .:53 {
        errors
        health {
           lameduck 5s
        }
        ready
        kubernetes cluster.local in-addr.arpa ip6.arpa {
           pods insecure
           fallthrough in-addr.arpa ip6.arpa
           ttl 30
        }
        prometheus :9153
        forward . /etc/resolv.conf
        cache 30
        loop
        reload
        loadbalance
    }
```
we should add `.cluster.local`

Reply:
yup, thanks for linking to the spec 👍 
I also found a couple of issues related to this: https://github.com/kubernetes/kubeadm/issues/2114

I think we'll need to add all of those since k8s also resolves partially qualified names
`.svc,.svc.cluster,.svc.cluster.local` especially for things like admission webhook which only use `.svc` by default https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#service-reference, and this still won't solve the problem where services should be resolvable through `http://<service>` and `http://<service>.<ns>`

Also, I wasn't sure if `cluster.local` is static or something that we should be picking up from the coredns config, but looks like it is the default.

Reply:
> I think we'll need to add all of those since k8s also resolves partially qualified names
> `.svc,.svc.cluster,.svc.cluster.local` especially for things like admission webhook which only use `.svc` by default https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#service-reference, and this still won't solve the problem where services should be resolvable through `http://<service>` and `http://<service>.<ns>`

TIL, thanks for the info

> Also, I wasn't sure if cluster.local is static or something that we should be picking up from the coredns config, but looks like it is the default.

it is not, one example is the kubernetes CI that changes it, but I think that for kind environments we should be safe, maybe just add a comment mentioning this, don't know , an environment with proxies and changing the default kind domain seems a very specific case